server/requeststojointeamhandler.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/fluidkeys/api/datastore"
+	"github.com/fluidkeys/api/v1structs"
+	"github.com/fluidkeys/crypto/openpgp"
+	"github.com/gofrs/uuid"
+	"github.com/gorilla/mux"
+)
+
+func listRequestsToJoinTeamHandler(w http.ResponseWriter, r *http.Request) {
+	apparentSignerKey, err := getAuthorizedUserPublicKey(r)
+	if err == errAuthKeyNotFound {
+		writeJsonError(w,
+			fmt.Errorf("public key that signed the roster has not been uploaded"),
+			http.StatusBadRequest)
+		return
+	} else if err != nil {
+		writeJsonError(w, err, http.StatusBadRequest)
+		return
+	}
+
+	teamUUID, err := uuid.FromString(mux.Vars(r)["teamUUID"])
+	if err != nil {
+		writeJsonError(w, err, http.StatusBadRequest)
+		return
+	}
+
+	dbTeam, err := datastore.GetTeam(nil, teamUUID)
+	if err == datastore.ErrNotFound {
+		writeJsonError(w, fmt.Errorf("team not found"), http.StatusNotFound)
+		return
+
+	} else if err != nil {
+		writeJsonError(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	var keyring openpgp.EntityList = []*openpgp.Entity{&apparentSignerKey.Entity}
+
+	_, err = openpgp.CheckArmoredDetachedSignature(
+		keyring,
+		strings.NewReader(dbTeam.Roster),
+		strings.NewReader(dbTeam.RosterSignature),
+	)
+
+	if err != nil {
+		writeJsonError(w, fmt.Errorf("signature verification failed"), http.StatusBadRequest)
+		return
+	}
+
+	var requestsToJoinTeam = []datastore.RequestToJoinTeam{}
+
+	err = datastore.RunInTransaction(func(txn *sql.Tx) error {
+		if exists, err := datastore.TeamExists(txn, teamUUID); err != nil {
```

Comment:
You already tested if the team exists with the previous use of `GetTeam`. I think you want to drop `TeamExists` and bring the `GetTeam` inside this transaction